File: src/SOP8/Base/Account/Base.Account.IBLL/Interface/IPasswordPolicy.cs
using System.Collections.Generic;

namespace Base.Account.IBLL.Interface
{
    public interface IPasswordPolicy
    {
        public bool NeedUpperCase
        {
            get; set;
        }

        public bool NeedLowerCase
        {
            get; set;
        }

        public bool NeedAlphabet
        {
            get; set;
        }

        public bool NeedNumber
        {
            get; set;
        }

        public bool NeedCharacter
        {
            get; set;
        }

        public bool NeedNumberOrCharacter
        {
            get; set;
        }

        public int? MinimumLength
        {
            get; set;
        }

        public int? MaximumLength
        {
            get; set;
        }

        public List<char> AllowCharacters
        {
            get; set;
        }
    }
}

File: src/SOP8/Egovframework/Base/base-account-ibll/src/main/java/egovframework/base/account/ibll/interface_/IPasswordPolicy.java
package egovframework.base.account.ibll.interface_;

import java.util.List;

public interface IPasswordPolicy {

    boolean isNeedUpperCase();
    void setNeedUpperCase(boolean needUpperCase);

    boolean isNeedLowerCase();
    void setNeedLowerCase(boolean needLowerCase);

    boolean isNeedNumber();
    void setNeedNumber(boolean needNumber);

    boolean isNeedCharacter();
    void setNeedCharacter(boolean needCharacter);

    Integer getMinimumLength();
    void setMinimumLength(Integer minimumLength);

    Integer getMaximumLength();
    void setMaximumLength(Integer maximumLength);

    List<Character> getAllowCharacters();
    void setAllowCharacters(List<Character> allowCharacters);
}